libs/patchwheel/parse.go

```diff
@@ -0,0 +1,63 @@
+package patchwheel
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// WheelInfo contains information extracted from a wheel filename
+type WheelInfo struct {
+	Distribution string   // Package distribution name
+	Version      string   // Package version
+	Tags         []string // Python tags (python_tag, abi_tag, platform_tag)
+}
+
+// CalculateNewVersion generates a new version string and filename based on the wheel info and modification time.
+// The version is updated according to the following rules:
+//   - if there is an existing part after + it is dropped
+//   - append +<mtime of the original wheel> to version
+func CalculateNewVersion(info *WheelInfo, mtime time.Time) (newVersion, newFilename string) {
+	baseVersion := strings.SplitN(info.Version, "+", 2)[0]
+
+	dt := strings.Replace(mtime.Format("20060102150405.00"), ".", "", 1)
+	dt = strings.Replace(dt, ".", "", 1)
+	newVersion = baseVersion + "+" + dt
+
+	newFilename = fmt.Sprintf("%s-%s-%s.whl",
+		info.Distribution,
+		newVersion,
+		strings.Join(info.Tags, "-"))
+
+	return newVersion, newFilename
+}
+
+// ParseWheelFilename parses a wheel filename and extracts its components.
+// Wheel filenames follow the pattern: {distribution}-{version}-{python_tag}-{abi_tag}-{platform_tag}.whl
```

Comment:
As per spec there might be 4th optional `build` tag which is not handled here
> The wheel filename is {distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl.

https://peps.python.org/pep-0491/

Reply:
Added.